cmd/zed/relationship.go

```diff
@@ -0,0 +1,105 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	api "github.com/authzed/authzed-go/arrakisapi/api"
+	"github.com/jzelinskie/cobrautil"
+	"github.com/jzelinskie/stringz"
+	"github.com/spf13/cobra"
+	"golang.org/x/term"
+)
+
+var relationshipCmd = &cobra.Command{
+	Use:   "relationship <subcommand>",
+	Short: "perform CRUD operations on the relationships in a permission system",
+}
+
+var createCmd = &cobra.Command{
+	Use:               "create <subject:id> relation <object:id>",
+	Short:             "create a relationship between a user and an object",
+	Args:              cobra.ExactArgs(3),
+	PersistentPreRunE: cobrautil.SyncViperPreRunE("ZED"),
+	RunE:              writeRelationshipCmdFunc(api.RelationTupleUpdate_CREATE),
+}
+
+var touchCmd = &cobra.Command{
+	Use:               "touch <subject:id> relation <object:id>",
+	Short:             "touch a relationship between a user and an object",
+	Args:              cobra.ExactArgs(3),
+	PersistentPreRunE: cobrautil.SyncViperPreRunE("ZED"),
+	RunE:              writeRelationshipCmdFunc(api.RelationTupleUpdate_TOUCH),
+}
+
+var deleteCmd = &cobra.Command{
+	Use:               "delete <subject:id> relation <object:id>",
+	Short:             "delete a relationship between a user and an object",
+	Args:              cobra.ExactArgs(3),
+	PersistentPreRunE: cobrautil.SyncViperPreRunE("ZED"),
+	RunE:              writeRelationshipCmdFunc(api.RelationTupleUpdate_DELETE),
+}
+
+func writeRelationshipCmdFunc(operation api.RelationTupleUpdate_Operation) func(cmd *cobra.Command, args []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		var userNS, userID string
+		err := stringz.SplitExact(args[0], ":", &userNS, &userID)
```

Comment:
Extract these into helper methods used here and in Check, and have it handle the `...` default stuff